src/platform_impl/web/event_loop/runner.rs

```diff
@@ -348,16 +347,11 @@ impl<T: 'static> Shared<T> {
                 &mut control,
             );
 
-            // Then we resize the canvas to the new size and send a `Resized` event:
+            // Then we resize the canvas to the new size, a new
+            // `Resized` event will be sent by the `ResizeObserver`:
```

Comment:
> This isn't necessarily true. When the scale factor changes, it's possible for the logical size to change but the physical size to stay the same (e.g., if the canvas takes up 100% of the window and the user zooms in), in which case the `ResizeObserver` won't trigger because it's only watching the physical size. This is why my implementation used two `ResizeObserver`s, one for each size.

I didn't understand why this was necessary in #2074, we already get a notification when the scale factor changes, how does this help us here?

The comment is correct, **if** the physical size of the canvas actually changes, we will get a notification, as far as I'm aware we don't really care if it's logical size changes.
> Note that you can't get the same `ResizeObserver` to observe both sizes of an element, when you tell it to observe the same element twice it cancels the first observation.

This is good to know, thank you.

Reply:
> The comment is correct, **if** the physical size of the canvas actually changes, we will get a notification, as far as I'm aware we don't really care if it's logical size changes.

That's true, I guess you don't really need to be notified if only the logical size changes. I think I was just playing it safe in #2074 and making sure that an event got sent any time it could possibly be warranted.

When you're observing the logical size of the canvas, though, it's also possible to miss a change in the _physical_ size. For instance, if the canvas has a fixed `width`/`height` set in CSS, zooming in will change the physical size but not the logical size, and so it won't get picked up. So, you probably need to add some code to `scale_factor_changed` which checks for that situation, and manually dispatches a `Resized` event if the `ResizeObserver` isn't going to trigger.

Reply:
> When you're observing the logical size of the canvas, though, it's also possible to miss a change in the _physical_ size.

You mean on Safari I assume? Yeah, good point.

Reply:
Done, I also addressed #2864 while I'm at it.